tests/Aspire.Hosting.Tests/AddParameterTests.cs

```diff
@@ -112,82 +112,142 @@ public void ParametersWithConfigurationValueDoNotGetDefaultValue()
         var expectedManifest = $$"""
             {
               "type": "parameter.v0",
-              "value": "{val1.inputs.value}",
+              "value": "{pass.inputs.value}",
               "inputs": {
                 "value": {
                   "type": "string"
                 }
               }
             }
             """;
-        Assert.Equal(expectedManifest, param1Manifest.ToString());
+        Assert.Equal(expectedManifest, paramManifest.ToString());
     }
 
-    [Fact]
-    public async Task ParametersWithDefaultValueObjectOverloadUsedRegardlessOfConfigurationValue()
+    [Theory]
+    // We test all the combinations of {direct param, callback param} x {config value, no config value}
+    [InlineData(false, false)]
+    [InlineData(false, true)]
+    [InlineData(true, false)]
+    [InlineData(true, true)]
+    public async Task ParametersWithDefaultValueGetPublishedIfPublishFlagIsPassed(bool useCallback, bool hasConfig)
     {
         var appBuilder = DistributedApplication.CreateBuilder();
 
-        appBuilder.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        if (hasConfig)
         {
-            ["Parameters:val1"] = "ValueFromConfiguration",
-        });
+            appBuilder.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+            {
+                ["Parameters:pass"] = "ValueFromConfiguration"
+            });
+        }
 
-        var genParam = new GenerateParameterDefault
+        if (useCallback)
         {
-            MinLength = 10,
-        };
-
-        // We have 2 params, one with a config value and one without. Both get a generated param default value.
-        var parameter1 = appBuilder.AddParameter("val1", genParam);
-        var parameter2 = appBuilder.AddParameter("val2", genParam);
+            appBuilder.AddParameter("pass", () => "DefaultValue", publishValueAsDefault: true);
+        }
+        else
+        {
+            appBuilder.AddParameter("pass", "DefaultValue", publishValueAsDefault: true);
+        }
 
         using var app = appBuilder.Build();
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+
+        // Make sure the code value is used, ignoring any config value
+        var parameterResource = Assert.Single(appModel.Resources.OfType<ParameterResource>(), r => r.Name == "pass");
+        Assert.Equal($"DefaultValue", parameterResource.Value);
+
+        // The manifest should include the default value, since we passed publishValueAsDefault: true
+        var paramManifest = await ManifestUtils.GetManifest(appModel.Resources.OfType<ParameterResource>().Single(r => r.Name == "pass"));
+        var expectedManifest = $$"""
+            {
+              "type": "parameter.v0",
+              "value": "{pass.inputs.value}",
+              "inputs": {
+                "value": {
+                  "type": "string",
+                  "default": {
```

Comment:
@mitchdenny This is new schema for parameters to specify a default value. We should update the schema and file an issue on azd to implement this.

@vhvb1989 Thoughts on implementing this idea of no prompting for parameters with defaults? Or being able to suppress them somehow?

Reply:
> @vhvb1989 Thoughts on implementing this idea of no prompting for parameters with defaults? Or being able to suppress them somehow?

If it's a 'default', there needs to be a way to override them. Maybe there are two modes (e.g. controlled by switches):
- Use all manifest defaults without prompting
- Prompt, but prefill the field with the manifest default

Reply:
FYI, azd suports a `generate` field in `default`, like

![image](https://github.com/user-attachments/assets/64a56074-c6a8-4ab4-9742-dddc88b38c2f)

Azd uses the gereate config to produce the value instead of prompt.  If you need a static default value, we could add a rule to the generate config to just use the provided value

@eerhardt 

When we added support for this, we considered a case where azd could ask forlk if they wanted to auto-generate the value (based on the rules) or enter a value manually.  This is supported by azd already (just not mapped to any configuration from the Aspire Manifest).

Reply:
To me, it feels a little strange to have this in `generate` (because there is no generation), which is why I went with an alternative `value` property, e.g.

```json
              "inputs": {
                "value": {
                  "type": "string",
                  "default": {
                    "value": "DefaultValue"
                  }
                }
              }
```

Either way, whoever owns the aspire manifest schema should decide where to put that. If we want it under `generate` that's easy to do.

Reply:
I agree that generate is strange. I like the above proposal of having a "default" section.

Reply:
The idea behind `generate` was that you could have either `value` or `generate`. If generate was present, then the fields in that sub-object would control the generation logic. So your proposal makes sense.

Reply:
Great, so then we are all good with the proposed schema change as it is in this PR? If so, what is the next step to get this in?

Reply:
I'll do a final pass

Reply:
Also, should I be modifying the schema file accordingly? I've never touched one of those, but I may as well learn how to do it!

Reply:
Yep

Reply:
Ok, I pushed the schema change. I also added test coverage for "generate", as it was missing. @mitchdenny PTAL

Reply:
@mitchdenny Also, how does the schema make its way from this repo to https://github.com/dotnet/dotnet/blob/main/src/aspire/src/Schema/aspire-8.0.json? Automatic or we need to push it?

Reply:
Manually I think

Reply:
Yeah all manual.